tactics/wheelhouse.py

```diff
@@ -0,0 +1,97 @@
+import logging
+
+from charmtools.build.tactics import ExactMatch, Tactic
+from charmtools import utils
+
+log = logging.getLogger(__name__)
+
+
+class WheelhouseTactic(ExactMatch, Tactic):
+    """
+    This tactic matches the file `wheelhouse.txt` and is called to process
+    that file.
+    """
+    kind = "dynamic"
+    FILENAME = 'wheelhouse.txt'
+
+    def __init__(self, *args, **kwargs):
+        super(WheelhouseTactic, self).__init__(*args, **kwargs)
+        self.tracked = []
+        self.previous = []
+
+    @property
+    def dest(self):
+        return self.target.directory / 'lib'
+
+    def __str__(self):
+        return "Building wheelhouse in {}".format(self.dest)
+
+    def combine(self, existing):
+        """
+        Maintain a list of the tactic instances for each previous layer's
+        `wheelhouse.txt` file, in order.
+        """
+        self.previous = existing.previous + [existing]
+        return self
+
+    def __call__(self):
+        """
+        Process the wheelhouse.txt file.
+
+        This gets called once on the tactic instance for the highest level
+        layer which contains a `wheelhouse.txt` file (e.g., the charm layer).
+        It then iterates the instances representing all of the previous
+        layers, in order, so that higher layers take precedence over base
```

Comment:
If it starts at the higher layer, won't lower layer files then overwrite the previously added higher layer files? So in face lower layers appear to take precedence?

Reply:
Oh, so the code from higher layers calls lower layers first. I think I just misunderstood the code comment